RedBeanPHP/OODBBean.php

```diff
@@ -537,14 +537,21 @@ public function import( $array, $selection = FALSE, $notrim = FALSE )
 		if ( is_string( $selection ) ) {
 			$selection = explode( ',', $selection );
 		}
-		if ( !$notrim && is_array( $selection ) ) {
-			foreach ( $selection as $key => $selected ) {
-				$selection[$key] = trim( $selected );
+		if ( is_array( $selection ) ) {
+			if ( $notrim ) {
+				$selected = array_flip($selection);
+			} else {
+				$selected = array();
+				foreach ( $selection as $key => $select ) {
+					$selected[trim( $select )] = TRUE;
```

Comment:
We can't do it using `array_map('trim', $array)`?

Reply:
I actually wanted to do something like this, but array_map is still slower than foreach in PHP7 so I went with foreach.

My first idea:

```php
if ( is_array( $selection ) ) {
    if ( !$notrim ) {
        $selection = array_map( 'trim', $selection );
    }
    $selected = array_flip( $selection );
} else {
    $selected = FALSE;
}
```

Reply:
Is there that much difference performance wise?

Reply:
The mean improvement is around 20% from my tests (PHP7) but it should be even greater with PHP 5.X.
If we're talking about 1 single operation, I agree that saving 2E-6 second isn't really interesting but on a very big script with lots of stuff to import I guess it could have a small impact.

Anyway, it isn't that important of a PR :)